subsys/net/ip/net_core.c

```diff
@@ -90,9 +99,12 @@ static inline enum net_verdict process_data(struct net_pkt *pkt)
 				net_stats_update_processing_error(
 							net_pkt_iface(pkt));
 			}
-
 			return ret;
 		}
+		if (!being_processed_by_correct_thread(pkt)) {
```

Comment:
like here, it would be preferred if there is empty line before the added `if` statement